web/src/app/pages/edit-survey/edit-job/edit-job.component.ts

```diff
@@ -95,6 +100,19 @@ export class EditJobComponent {
         )
       )
     );
+
+    this.lois = await firstValueFrom(
+      this.loiService.getLocationsOfInterest$().pipe(
+        map(lois => LocationOfInterestService.getLoisWithNames(lois)),
+        map(lois =>
+          List(
+            lois
+              .toArray()
+              .filter((loi: LocationOfInterest) => loi.jobId === this.jobId)
```

Comment:
This seems to be a roundabout way of getting the lois for he current job. Can we just expose a method in the service to do this? It could later be optimized there in the data layer if necessary.